File: src/main/java/org/example/momento/Editor.java
package org.example.momento;

public class Editor {
    private String content;

    public String getContent() {
        return content;
    }
    public EditorState createState(){
        return new EditorState(content);
    }
    public void restore(EditorState editorState){
        content = editorState.getContent();
    }

    public void setContent(String content){
        this.content = content;
    }

    @Override
    public String toString() {
        return "Editor{" +
                "content='" + content + '\'' +
                '}';
    }
}

File: Kotlin/src/momento/Editor.kt
package momento

class Editor
{
    lateinit var content : String

    fun getContent():String {
        return content
    }
    fun createState():EditorState{
        return EditorState(content)
    }

    fun restore(content:String){
        this.content = content
    }

    fun setContent(content:String) {
        this.content = content
    }

}